taskvine/src/manager/vine_manager.c

```diff
@@ -4251,36 +4265,18 @@ char *vine_monitor_wrap(struct vine_manager *q, struct vine_worker_info *w, stru
 	return wrap_cmd;
 }
 
-static double vine_task_priority(void *item)
-{
-	assert(item);
-	struct vine_task *t = item;
-	return t->priority;
-}
-
 /* Put a given task on the ready list, taking into account the task priority and the manager schedule. */
 
-static void push_task_to_ready_list(struct vine_manager *q, struct vine_task *t)
+static void push_task_to_ready_tasks(struct vine_manager *q, struct vine_task *t)
 {
-	int by_priority = 1;
-
 	if (t->result == VINE_RESULT_RESOURCE_EXHAUSTION) {
 		/* when a task is resubmitted given resource exhaustion, we
 		 * push it at the head of the list, so it gets to run as soon
 		 * as possible. This avoids the issue in which all 'big' tasks
 		 * fail because the first allocation is too small. */
-		by_priority = 0;
-	}
-
-	if (by_priority) {
-		/*If a task has a priority of 0 it gets added to the end of the ready list.*/
-		if (vine_task_priority(t) != 0) {
-			list_push_priority(q->ready_list, vine_task_priority, t);
-		} else {
-			list_push_tail(q->ready_list, t);
-		}
+		priority_queue_push(q->ready_tasks, t, DBL_MAX);
```

Comment:
No, this isn't right.  You don't want to change the priority of the task to "best possible".   You just want it to be considered first among tasks of equal priority.  Perhaps you need an operation to that effect on the queue.